eslint-rules/ts-no-use-before-define.js

```diff
@@ -0,0 +1,288 @@
+// Added here until https://github.com/nzakas/eslint-plugin-typescript/pull/34
```

Comment:
[It doesn’t seem to catch this](https://www.typescriptlang.org/play/index.html#src=foo%0D%0Avar%20foo%20%3D%20'bar').